#! /usr/bin/env python
PACKAGE='golfcart_push'

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name          Type   Level 		Description               Default   Min   Max
gen.add("h_center",   int_t, 0,       "Hue threshold center",   120,      0,    180)
gen.add("h_width",    int_t, 0,       "Hue threshold width",    10,       1,    90)
gen.add("s_thres",    int_t, 0,       "Saturation threshold",   150,      0,    255)
gen.add("v_thres",    int_t, 0,       "Value threshold",        200,      0,    255)


#       Name                        Type       Lvl    Description                                          Default     Min     Max
gen.add("x_min",                    double_t,  0,     "Region of interest",                                0.0,       -100.0,    0.0)
gen.add("x_max",                    double_t,  0,     "Region of interest",                                100.0,      0.0,     100.0)
gen.add("y_min",                    double_t,  0,     "Region of interest",                                -100.0,     -100.0,   0.0)
gen.add("y_max",                    double_t,  0,     "Region of interest",                                1000.0,      0.0,     100.0)
gen.add("z_min",                    double_t,  0,     "Region of interest",                                -1000,      -1000.0,   0.0)
gen.add("z_max",                    double_t,  0,     "Region of interest",                                1000,         0.0,     1000.0)
gen.add("voxel_size",               double_t,  0,     "Downsampling voxel size",                           0.3,       0.01,    1.0)
gen.add("cluster_tol",              double_t,  0,     "Distance to include point in cluster",              1,         0.001,   10.0)
gen.add("min_cluster_size",         int_t,     0,     "Minimum allowed number of points in a cluster",     3,         1,       1000)
gen.add("max_cluster_size",         int_t,     0,     "Maximum allowed number of points in a cluster",     500,       50,      1000)

exit(gen.generate(PACKAGE, PACKAGE, "GolfcartPush"))
